lib/immobilienscout/response_parser.rb

```diff
@@ -33,13 +36,50 @@ def code
     end
 
     def messages(response_body)
-      return response_body unless response_body['common.messages']
+      if response_body['common.messages']
```

Comment:
Maybe you can use guard clauses here:
```ruby
return common_message(response_body) if response_body['common.messages']
return on_top_placement_message(response_body, on_top_placement_type(response_body)) if on_top_placement_type(response_body)

response_body
``` 

If for some reason the lines are too big you can save store the logic in a separated method